From 070414dd4155e13583e5e8b16bed1a5b68d32910 Mon Sep 17 00:00:00 2001
From: Heng Liu <liuhenga@uniontech.com>
Date: Fri, 25 Oct 2024 10:17:45 +0800
Subject: [PATCH] fix crash when attach differ shellsurface to the same
 shellsurfaceitem

Change shellSurface from a raw pointer to a QPointer to prevent
crash caused by accessing the wild pointer when shellSurface is destroyed

Change-Id: Iabf25ce9e3fab416b5b927bf768e94398516a710
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
(cherry picked from commit 1f73059ec58b1e9ab137b701607b59cfa77750c9)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/compositor/extensions/qwaylandquickshellsurfaceitem_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor/extensions/qwaylandquickshellsurfaceitem_p.h b/src/compositor/extensions/qwaylandquickshellsurfaceitem_p.h
index 46e5f65a5..10b10a0c2 100644
--- a/src/compositor/extensions/qwaylandquickshellsurfaceitem_p.h
+++ b/src/compositor/extensions/qwaylandquickshellsurfaceitem_p.h
@@ -41,7 +41,7 @@ class Q_WAYLANDCOMPOSITOR_EXPORT QWaylandQuickShellSurfaceItemPrivate : public Q
     void lower() override;
 
     QWaylandQuickShellIntegration *m_shellIntegration = nullptr;
-    QWaylandShellSurface *m_shellSurface = nullptr;
+    QPointer<QWaylandShellSurface> m_shellSurface = nullptr;
     QQuickItem *m_moveItem = nullptr;
     bool m_autoCreatePopupItems = true;
     bool staysOnTop = false;
